multipers/filtration/filtrations.py

```diff
@@ -147,9 +149,86 @@ def DegreeRips(*, points = None, distance_matrix=None, ks=None, threshold_radius
 
+    Input:
+     - points: The point cloud as an ArrayLike of shape (n, d) where n is the number of points and d is the dimension of the points.
+     - beta: The beta parameter for the Delaunay Core Bifiltration (default 1.0).
+     - k_max: The maximum number of nearest neighbors to consider (default None). If None or greater than the number of points in the point cloud, k_max is set to the number of points in the point cloud.
+     - k_step: The step size for the number of nearest neighbors (default 1).
+     - precision: The precision of the computation of the AlphaComplex, one of ['safe', 'exact', 'fast'] (default 'safe'). See the GUDHI documentation for more information.
+     - verbose: Whether to print progress messages (default False).
+     - max_alpha_square: The maximum squared alpha value to consider when createing the alpha complex (default inf). See the GUDHI documentation for more information.
+    """
+    if k_max is None or k_max > len(points):
+        k_max = len(points)
+
+    assert len(points) > 0, f"The point cloud must contain at least one point."
+    assert points.ndim == 2, f"The point cloud must be a 2D array, got {points.ndim}D."
+    assert beta >= 0, f"The parameter beta must be positive, got {beta}."
+    assert k_max > 0, f"The parameter k_max must be positive, got {k_max}."
+    assert k_step > 0, f"The parameter k_step must be positive, got {k_step}."
+    assert precision in ["safe", "exact", "fast"], (
+        f"The parameter precision must be one of ['safe', 'exact', 'fast'], got {precision}."
+    )
+    ks = np.arange(1, k_max + 1, k_step)
+    if verbose:
+        print(
+            f"Computing the Delaunay Core Bifiltration of {len(points)} points in dimension {points.shape[1]} with parameters:"
+        )
+        print(f"\tbeta = {beta}")
+        print(f"\tk_max = {k_max}")
+        print(f"\tk_step = {k_step} (total of {len(ks)} k-values)")
+
+    if verbose:
+        print("Building the alpha complex...")
+    alpha_complex = gd.AlphaComplex(
+        points=points, precision=precision
+    ).create_simplex_tree(max_alpha_square=max_alpha_square)
+
+    if verbose:
+        print("Computing the k-nearest neighbor distances...")
+    knn_distances = KDTree(points).query(points, k=ks)[0]
+
+    max_dim = alpha_complex.dimension()
+    vertex_arrays_in_dimension = [[] for _ in range(max_dim + 1)]
+    squared_alphas_in_dimension = [[] for _ in range(max_dim + 1)]
+    for simplex, alpha_squared in alpha_complex.get_simplices():
+        dim = len(simplex) - 1
+        squared_alphas_in_dimension[dim].append(alpha_squared)
+        vertex_arrays_in_dimension[dim].append(simplex)
+
+    alphas_in_dimension = [np.sqrt(np.array(alpha_squared, dtype=np.float64)) for alpha_squared in squared_alphas_in_dimension]
+    vertex_arrays_in_dimension = [np.array(vertex_array, dtype=int) for vertex_array in vertex_arrays_in_dimension]
```

Comment:
Small optimization, the `insert_batch` will copy this to `int32` otherwise   
```suggestion
    vertex_arrays_in_dimension = [np.array(vertex_array, dtype=np.int32) for vertex_array in vertex_arrays_in_dimension]
```